src/modules/commander/Arming/PreFlightCheck/checks/failureDetectorCheck.cpp

```diff
@@ -45,17 +45,30 @@ bool PreFlightCheck::failureDetectorCheck(orb_advert_t *mavlink_log_pub, const v
 
 	if (status.failure_detector_status != vehicle_status_s::FAILURE_NONE) {
 		if (report_fail) {
+			// in case an error is forgotten to get handled here the user at least gets a less specific message
+			const char *message = "Unspecified";
+
 			if (status.failure_detector_status & vehicle_status_s::FAILURE_ROLL) {
-				mavlink_log_critical(mavlink_log_pub, "Preflight Fail: Roll failure detected");
+				message = "Roll";
```

Comment:
This is fine, but in general I think I'd prefer to not be manipulating strings at all. We need to move beyond that entirely for a number of reasons (events interface, localization, flash usage, etc).

Reply:
More specifically one reason it's preferable to leave each string together as a single literal is for the possibility to tokenize later and replace the actual string literal with a more compact binary token.

Reply:
I see, I actually did this to safe flash since the strings contain quite some redundancy. I'm happy to follow your suggestions. I can change it back if you like.

Reply:
I changed it in "failureDetectorCheck: remove string manipulation". It uses 168Bytes more flash :man_shrugging: